### PR TITLE
Prometheus terraform modules

### PR DESCRIPTION
**Pull request by saliceti**
## What
Terraform modules to deploy all the required prometheus components to monitor paas.

## How to review
This can be demoed along the PR on https://github.com/DFE-Digital/teacher-vacancy-service/

### CODE REVIEW

**Comment by vigneshmsft** [comment]
nice, quite a big one 👏🏼
will review tomorrow.

File: terraform/modules/prometheus/templates/prometheus.yml.tmpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+global:
+  scrape_interval: 10s
+
+scrape_configs:
+- job_name: teaching-vacancies-prometheus-exporter-dfe-teacher-services
```
Comment (vigneshmsft) [inline terraform/modules/prometheus/templates/prometheus.yml.tmpl]: can this be passed from parmeters as well? keeping the template generic

File: terraform/modules/prometheus/templates/alert.rules.tmpl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+groups:
+  - name: example
+    rules:
+    - alert: HighCPU
+      expr: avg by (space) (cpu{space=~"teaching-vacancies-.+"}) > 0
```
Comment (vigneshmsft) [inline terraform/modules/prometheus/templates/alert.rules.tmpl]: same as the other comment, can space name be generic in the template?

File: terraform/modules/prometheus/templates/prometheus.yml.tmpl (hunk 1 above)
Comment (saliceti) [reply terraform/modules/prometheus/templates/prometheus.yml.tmpl]: done, thanks

File: terraform/modules/prometheus/templates/alert.rules.tmpl (hunk 2 above)
Comment (saliceti) [reply terraform/modules/prometheus/templates/alert.rules.tmpl]: done, the default is emtpy config now

**Review by vigneshmsft — APPROVED** [review]

**Pull request merged**